### PR TITLE
Feature/cit 304/migrate connect and drop TEdgeConfig

### PR DESCRIPTION
**Pull request by mneumann**
- Migrates the `tedge connect` command
- And does a lot of refactoring
- And finally get rid of old TEdgeConfig

### CODE REVIEW

File: tedge/src/cli/connect/bridge_config.rs [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+    pub bridge_certfile: FilePath,
+    pub bridge_keyfile: FilePath,
+    pub use_mapper: bool,
+    pub try_private: bool,
+    pub start_type: String,
+    pub clean_session: bool,
+    pub notifications: bool,
+    pub bridge_attempt_unsubscribe: bool,
+    pub topics: Vec<String>,
+}
```
Comment (makr11st) [inline tedge/src/cli/connect/bridge_config.rs:24]: As we are using strong types for the `BridgeConfigAzureParams` and `BridgeConfigC8yParams` shouldn't we also use strong types here?
Eg:
* pub address: String, -> pub address: Url,
* pub config_file: String -> pub config_file: PathBuf / Buf / FilePath
* pub cloud_name: String ->  pub cloud_name: Cloud

**Review by makr11st — APPROVED** [review]
Looks good. Just the types in the top level bridgeconfig which could be updated, but don't have to.

File: tedge/src/cli/connect/bridge_config.rs (hunk 1 above)
Comment (mneumann) [reply tedge/src/cli/connect/bridge_config.rs:24]: Very good point! We definitively should!

One issue I came across when trying to convert `address` to an `Url` was that the way we use `Url` is slightly wrong. See 

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d9b19f34c756778ddd86ce026c874a9e

When parsing `test.test.io:8883`, we actually treat `test.test.io` as schema and `8883` as path. 

About `config_file`... that's a good point, too. This is used as the name of the config file, not the full path to it. I was concerned that changing that to a `FilePath` etc. people might confuse it with the full path.

`Cloud_name`. Yes, you are right again. 

We should add more precise types.

Comment (mneumann) [reply tedge/src/cli/connect/bridge_config.rs:24]: @makr11st I'd be much much better if we could move the `Url::parse` to where we create the `BridgeConfig`. We still need the validation, but that would only check if the files actually exist. As it isn't that simple as it seems, I'd like to keep that for a possible future task :)

Comment (makr11st) [reply tedge/src/cli/connect/bridge_config.rs:24]: I think I exactly know what you mean as I was dealing with the same problem, `Url::parse` is correct here as it adheres to relevant url spec: https://url.spec.whatwg.org/ .

**Comment by mneumann** [comment]
@makr11st rebased against `main` (your last commit) and get rid of old `TEdgeConfig`. Please review last 3 commits again.

**Comment by makr11st** [comment]
> @makr11st rebased against `main` (your last commit) and get rid of old `TEdgeConfig`. Please review last 3 commits again.

LGTM

**Review by didier-wenzek — APPROVED** [review]
Approved!

I agree that the types might be improved for the bridge configuration. But this needs to be done in a separate PR. This one is already huge enough.

**Pull request merged**